manager/manager.go

```diff
@@ -336,6 +338,18 @@ func (m *Manager) issue(ctx context.Context, volumeID string) error {
 		return fmt.Errorf("waiting for request: %w", err)
 	}
 
+	// Default the renewal time to be halfway through the certificate's duration.
+	// The implementation's writeKeypair function may override this value before
+	// writing to the storage layer.
+	block, _ := pem.Decode(req.Status.Certificate)
+	crt, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return fmt.Errorf("parsing issued certificate: %w", err)
+	}
+	duration := crt.NotAfter.Sub(crt.NotBefore)
+	midpoint := crt.NotBefore.Add(duration / 2)
```

Comment:
This is different to other implementations across cert-manager, which default to `2/3`rds. Is the rational that csi-lib certs are expected to be shorter lived so there is less wriggle room during renewal?

Reply:
There wasn't particularly clear rationale for 1/2 vs 1/3rd nope. In pretty much all cases this _should_ be set by the `writeKeypair` function, so this should always be overridden. This is more of a safety net.. I'm pretty indifferent either way as to what we default this to, so can change it over to 2/3rds of the way through its life if preferred :)

Reply:
Yes please- just to have consistency and rule of least surprise.

Reply:
Done